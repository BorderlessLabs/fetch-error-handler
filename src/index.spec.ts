import { Request } from "cross-fetch";
import * as boom from "@hapi/boom";
import * as httpErrors from "http-errors";
import { errorHandler } from "./index";

describe("error handler", () => {
  describe("default accepts", () => {
    const req = new Request("/");
    const handler = errorHandler(req);

    it("should fail gracefully with non-error", async () => {
      const res = handler("test");

      expect(await res.text()).toMatchSnapshot();
    });

    it("should fail gracefully with empty error", async () => {
      const res = handler(undefined);

      expect(await res.text()).toMatchSnapshot();
    });

    it("should render an error", async () => {
      const res = handler(new Error("boom!"));

      expect(await res.text()).toMatchSnapshot();
    });

    it("should render boom status errors", async () => {
      const res = handler(boom.badRequest("data has an issue"));

      expect(await res.text()).toMatchSnapshot();
    });

    it("should render http errors status error", async () => {
      const res = handler(new httpErrors.BadRequest("data has an issue"));

      expect(await res.text()).toMatchSnapshot();
    });
  });

  describe("accept html", () => {
    const req = new Request("/", { headers: { accept: "text/html" } });
    const handler = errorHandler(req);

    it("should fail and return html", async () => {
      const res = handler(new Error("boom!"));

      expect(res.headers.get("content-type")).toEqual("text/html");
      expect(await res.text()).toMatchSnapshot();
    });
  });

  describe("accept json", () => {
    const req = new Request("/", { headers: { accept: "application/json" } });
    const handler = errorHandler(req);

    it("should fail and return json", async () => {
      const res = handler(new Error("boom!"));

      expect(res.headers.get("content-type")).toEqual("application/json");
      expect(await res.text()).toMatchSnapshot();
    });
  });
});
