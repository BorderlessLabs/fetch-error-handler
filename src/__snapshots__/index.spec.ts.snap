// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error handler accept html should fail and return html 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Error</title></head><body><pre>{
 &nbsp;&quot;status&quot;: 500,
 &nbsp;&quot;error&quot;: &quot;500 Error&quot;,
 &nbsp;&quot;message&quot;: &quot;Error&quot;
}</pre></body></html>"
`;

exports[`error handler accept json should fail and return json 1`] = `"{\\"status\\":500,\\"error\\":\\"500 Error\\",\\"message\\":\\"Error\\"}"`;

exports[`error handler default accepts should fail gracefully with empty error 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Error</title></head><body><pre>{
 &nbsp;&quot;status&quot;: 500,
 &nbsp;&quot;error&quot;: &quot;500 Error&quot;,
 &nbsp;&quot;message&quot;: &quot;Error&quot;
}</pre></body></html>"
`;

exports[`error handler default accepts should fail gracefully with non-error 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Error</title></head><body><pre>{
 &nbsp;&quot;status&quot;: 500,
 &nbsp;&quot;error&quot;: &quot;500 Error&quot;,
 &nbsp;&quot;message&quot;: &quot;Error&quot;
}</pre></body></html>"
`;

exports[`error handler default accepts should render an error 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Error</title></head><body><pre>{
 &nbsp;&quot;status&quot;: 500,
 &nbsp;&quot;error&quot;: &quot;500 Error&quot;,
 &nbsp;&quot;message&quot;: &quot;Error&quot;
}</pre></body></html>"
`;

exports[`error handler default accepts should render boom status errors 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Error</title></head><body><pre>{
 &nbsp;&quot;statusCode&quot;: 400,
 &nbsp;&quot;error&quot;: &quot;Bad Request&quot;,
 &nbsp;&quot;message&quot;: &quot;data has an issue&quot;
}</pre></body></html>"
`;

exports[`error handler default accepts should render http errors status error 1`] = `
"<!doctype html><html lang=\\"en\\"><head><meta charset=\\"utf-8\\"><title>Error</title></head><body><pre>{
 &nbsp;&quot;status&quot;: 400,
 &nbsp;&quot;error&quot;: &quot;400 Error&quot;,
 &nbsp;&quot;message&quot;: &quot;Error&quot;
}</pre></body></html>"
`;
